lib/rules/no-jquery.js

```diff
@@ -27,62 +31,85 @@ module.exports = {
   ERROR_MESSAGE,
 
   create(context) {
-    let destructuredAssignment;
-    let emberImportAliasName;
-    let jqueryImportAliasName;
-    const report = function (node) {
+    let report = function (node) {
       context.report(node, ERROR_MESSAGE);
+      // Only report this error message once
+      report = () => {};
     };
 
-    return {
-      ImportDeclaration(node) {
-        emberImportAliasName = ember.getEmberImportAliasName(node);
+    /**
+     * Analyze the scope of a jQuery identifier node
+     *
+     * @param  {Node} node jQuery identifier node
+     */
+    function handleJqueryIdentifier(node) {
+      // Find scope of identifier
+      let scopeVariable = findVariable(context.getScope(), node);
 
-        if (!jqueryImportAliasName && node.source && node.source.value === 'jquery') {
-          jqueryImportAliasName = node.specifiers[0].local.name;
-        }
-      },
+      // Find scope of identifier that is part of a member expression
+      // eg; Ember.$
+      if (isMemberExpression(node.parent)) {
+        scopeVariable = findVariable(context.getScope(), node.parent.object);
+      }
 
-      VariableDeclarator(node) {
-        if (emberImportAliasName) {
-          destructuredAssignment =
-            node.init && types.isMemberExpression(node.init)
-              ? node.id.name
-              : // assignment of type const $ = Ember.$;
-                utils
-                  .collectObjectPatternBindings(node, {
-                    [emberImportAliasName]: ['$'],
-                  })
-                  .pop();
+      // If identifier has no scope, assume it is a global variable
+      if (scopeVariable === null) {
+        if (isMemberOfThis(node.parent)) {
+          report(node.parent);
         }
-      },
 
-      MemberExpression(node) {
-        if (
-          (node.object.name === 'Ember' ||
-            node.object.name === 'Em' ||
-            (emberImportAliasName && node.object.name === emberImportAliasName)) &&
-          node.property.name === '$'
-        ) {
-          report(node);
+        if (isMemberOfGlobal(node.parent)) {
+          report(node.parent);
         }
 
-        if (jqueryImportAliasName && node.object.name === jqueryImportAliasName) {
-          report(node);
+        if (isGlobalCallExpression(node.parent)) {
+          report(node.parent);
         }
-      },
+      } else {
+        scopeVariable.defs.forEach((definition) => {
+          if (isParameterDefinition(definition)) {
```

Comment:
This means we can ignore any variables in scope that are function parameters. This was the main  issue I was hoping to fix..